server/src/main/java/org/opensearch/wlm/cancellation/HighestResourceConsumingTaskFirstSelectionStrategy.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.wlm.cancellation;
+
+import org.opensearch.wlm.QueryGroupTask;
+import org.opensearch.wlm.ResourceType;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+
+import static org.opensearch.wlm.cancellation.TaskCanceller.MIN_VALUE;
+
+/**
+ * Represents the highest resource consuming task first selection strategy.
+ */
+public class HighestResourceConsumingTaskFirstSelectionStrategy implements TaskSelectionStrategy {
+
+    private final Supplier<Long> nanoTimeSupplier;
+
+    public HighestResourceConsumingTaskFirstSelectionStrategy() {
+        this(System::nanoTime);
+    }
+
+    public HighestResourceConsumingTaskFirstSelectionStrategy(Supplier<Long> nanoTimeSupplier) {
+        this.nanoTimeSupplier = nanoTimeSupplier;
+    }
```

Comment:
Can we keep this constructor package-private if needed only tests?

Reply:
This will be accessed in Node.java during instantiation of QueryGroupService so this needs to be public